plugins/route/allocation/simple/plugin.go

```diff
@@ -50,20 +48,23 @@ func (p *SimpleAllocationPlugin) Allocate(route *routeapi.Route) (*routeapi.Rout
 // namespace (if provided) and the router shard dns suffix.
 func (p *SimpleAllocationPlugin) GenerateHostname(route *routeapi.Route, shard *routeapi.RouterShard) string {
 
-	name := route.ServiceName
+	name := route.Name
+	if len(name) == 0 {
+		name = route.ServiceName
+	}
 	if len(name) == 0 {
 		name = uuid.NewUUID().String()
 		glog.V(4).Infof("No service name passed, using generated name: %s", name)
 	}
 
 	s := ""
-	if len(route.Namespace) <= 0 {
+	if len(route.Namespace) == 0 {
 		s = fmt.Sprintf("%s.%s", name, shard.DNSSuffix)
 	} else {
 		s = fmt.Sprintf("%s.%s.%s", name, route.Namespace, shard.DNSSuffix)
 	}
 
-	glog.V(4).Infof("Generated hostname=%s for Route: %s", s, route.ServiceName)
+	glog.V(4).Infof("Generated hostname=%s for Route: %s", s, route.Name)
```

Comment:
If the route has an empty name it can't be save anyway.  You must specify
the name previous to this flow (unless you're describing something else).

On Jul 19, 2015, at 8:34 AM, Michail Kargakis <notifications@github.com>
wrote:

In plugins/route/allocation/simple/plugin.go
<https://github.com/openshift/origin/pull/3772#discussion_r34957877>:

>  		s = fmt.Sprintf("%s.%s", name, shard.DNSSuffix)
>  	} else {
>  		s = fmt.Sprintf("%s.%s.%s", name, route.Namespace, shard.DNSSuffix)
>  	}
>
> -	glog.V(4).Infof("Generated hostname=%s for Route: %s", s, route.ServiceName)
> +	glog.V(4).Infof("Generated hostname=%s for Route: %s", s, route.Name)

Shouldn't name propagate to route.Name if the latter is not set?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3772/files#r34957877>.